packages/nextra-theme-blog/src/mdx-theme.tsx

```diff
@@ -45,7 +45,6 @@ const HeaderLink = ({
   const slug = slugger.slug(ReactDOMServer.renderToStaticMarkup(children))
   return (
     <Tag {...props}>
-      <span className="subheading-anchor" id={slug} />
       <a href={'#' + slug} className="subheading">
```

Comment:
One thing I noticed that `slug` variable changes every time page reload after change but `props.id` stays the same.